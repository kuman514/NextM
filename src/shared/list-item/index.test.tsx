import { render, screen } from '@testing-library/react';

import ListItem from '.';

jest.mock(
  'next/link',
  () =>
    function MockedLink(props: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
      return <a {...props} />;
    }
);

describe('ListItem', () => {
  it('should render as expected (current: thumbnail, name, description)', () => {
    const { container } = render(
      <ListItem
        name="Iron Man"
        description="I am Iron Man, not Captain America."
        href="/character/1234"
        imageAlt="Iron Man"
        imageSrc="https://iron.man/captain_america.jpg"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render image if it does not have thumbnail', () => {
    const { container } = render(
      <ListItem
        name="Iron Man"
        description="I am Iron Man, not Captain America."
        href="/character/1234"
        imageAlt="Iron Man"
        imageSrc={null}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should has href=/character/1234 when the characterData is having id=1234', async () => {
    render(
      <ListItem
        name="Iron Man"
        description="I am Iron Man, not Captain America."
        href="/character/1234"
        imageAlt="Iron Man"
        imageSrc="https://iron.man/captain_america.jpg"
      />
    );

    const link = await screen.findByLabelText('list-item-link');
    expect((link as HTMLAnchorElement).href).toStrictEqual(
      'http://localhost/character/1234'
    );
  });
});
