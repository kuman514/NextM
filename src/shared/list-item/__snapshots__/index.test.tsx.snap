// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render as expected (current: thumbnail, name, description) 1`] = `
<li>
  <a
    aria-label="list-item-link"
    class="flex flex-col w-48 h-72 rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
    href="/character/1234"
  >
    <div
      class="w-48 h-48"
    >
      <img
        alt="Iron Man"
        class="w-full h-full object-cover"
        data-nimg="1"
        decoding="async"
        height="192"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Firon.man%2Fcaptain_america.jpg&w=384&q=75"
        srcset="/_next/image?url=https%3A%2F%2Firon.man%2Fcaptain_america.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Firon.man%2Fcaptain_america.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="192"
      />
    </div>
    <div
      class="flex flex-col justify-center items-center w-full h-36"
    >
      <span
        class="text-white"
      >
        Iron Man
      </span>
      <span
        class="text-white"
      >
        I am Iron Man, not Captain America.
      </span>
    </div>
  </a>
</li>
`;
